manifests/istio-telemetry/kiali/templates/kiali-kialicr.yaml

```diff
@@ -0,0 +1,118 @@
+{{- if .Values.kiali.enabled }}
+apiVersion: kiali.io/v1alpha1
+kind: Kiali
+metadata:
+  name: kiali
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: kiali
+    release: {{ .Release.Name }}
+annotations:
+  ansible.operator-sdk/verbosity: "3"
+spec:
+{{- if .Values.kiali.fullSpec }}
+{{ toYaml .Values.kiali.fullSpec | indent 2 }}
+{{- else }}
+  ###################################################################
+  # For details about the supported settings,
+  # see https://github.com/kiali/kiali/blob/master/operator/deploy/kiali/kiali_cr.yaml"
```

Comment:
it would be pretty cool if you could document this (possibly as we do in Istio, via a protobuf - and then generate the CR from that). We have tools to generate protobuf docs on istio.io.

Reply:
We do not have any protobuf integrations anywhere in Kiali, but are not opposed to it if it makes sense and makes life easier. Would need some community contributions for this; not sure what this would entail to implement this -- note that this Kiali CR is NOT consumed anywhere in Kiali via Go code - it is consumed by Ansible playbooks within the Operator.

Reply:
Good to hear the Ansible Operator work is being used here. From a docs perspective, we have tooling to process protobufs which is the only reason I mentioned it at all.

Processing yaml is harder, because no yaml parser I've been able to find round trips comments (or if they claim this feature, it doesn't work). It isn't Istio's mission to write a yaml round-trip parser :) It really isn't our mission to develop protobuf documentation tooling, but that has been done. I have heard there is an influx of interns coming - perhaps we can let them tackle this problem? @ericvn mentioned the interns to me a few days ago.